src/daemon/client.c

```diff
@@ -110,6 +110,19 @@ client_handle_set_configuration(struct lldpd *cfg, enum hmsg_type *type,
 		cfg->g_config.c_paused = config->c_paused;
 		levent_send_now(cfg);
 	}
+	/* reset */
+	if (CHANGED(c_reset) &&
+		config->c_reset == 1) {
+		log_debug("rpc", "client asked to reset lldpd's config");
+		cfg->g_config.c_reset = 1;	//mark that reset is being processed by daemon
+									//so that client won't accept any new reset
```

Comment:
You don't really need an "in progress" marker. lldpd is totally "evented", so nothing will execute in parallel until the end of this function. So, nobody will be able to observer a case where `cfg->g_onfig.c_reset == 1`. You can remove that and some other code on the client side about that.